### PR TITLE
[WIP] sys_vm fixes

### PR DESCRIPTION
**Pull request by isJuhn**
Fixes Doom 3: BFG Edition [BLES01679], [BLUS31024] and [NPUB30889]
Nothing -> Ingame
rpcs3_2018-05-16_02-15-24
Thanks to Deminating for the testing and the screenshot.

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_vm.cpp [hunk 1]
```diff
@@ -62,6 +62,11 @@ error_code sys_vm_append_memory(u32 addr, u32 size)
 {
 	sys_vm.warning("sys_vm_append_memory(addr=0x%x, size=0x%x)", addr, size);
 
+	if (vm::get(vm::memory_location_t::any, addr)->size + size > 0x10000000)
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_vm.cpp]: vm::memory_location_t::any can also be written as vm::any

**Comment by TigiHof** [comment]
Very cool! Works and runs with 60 fps, thanx a lot!

**Comment by MastaG** [comment]
Then fix looks straightforward.. merge it!

**Comment by AniLeo** [comment]
Nope. Unless you want to provide code review and/or justify why it's correct, wait on Nekotekina to review and merge.

**Comment by Relsig** [comment]
@isjuhn I meant to mention your original commit caused some missing assets in persona 5, especially in central street (missing signs, lights, etc) and allowed you to walk through some walls.

The rework fixed all those issues as far as I can tell and I haven't noticed any other weird happenings as of yet. I'll try to let you know if I spot something that isn't broken on master.

File: rpcs3/Emu/Cell/lv2/sys_vm.cpp [hunk 2]
```diff
@@ -21,13 +27,21 @@ error_code sys_vm_memory_map(u32 vsize, u32 psize, u32 cid, u64 flag, u64 policy
 	}
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/sys_vm.cpp]: First take the memory, then return it in the case vm::find_map failed

**Comment by jobs-git** [comment]
@isJuhn what happened to this pr?

**Comment by Cereal-Killa** [comment]
Up.

**Comment by AniLeo** [comment]
This is not a forum thread, stop flooding

**Pull request closed without merging**